supertokens_python/recipe/dashboard/recipe.py

```diff
@@ -0,0 +1,197 @@
+    InputOverrideConfig,
+    validate_and_normalise_user_input,
+    get_api_if_matched,
+    is_api_path,
+)
+
+
+class DashboardRecipe(RecipeModule):
+    recipe_id = "dashboard"
+    __instance = None
+
+    def __init__(
+        self,
+        recipe_id: str,
+        app_info: AppInfo,
+        api_key: str,
+        override: Union[InputOverrideConfig, None] = None,
+    ):
+        super().__init__(recipe_id, app_info)
+        self.config = validate_and_normalise_user_input(
+            api_key,
+            override,
+        )
+        recipe_implementation = RecipeImplementation(Querier.get_instance(recipe_id))
+        self.recipe_implementation = (
+            recipe_implementation
+            if self.config.override.functions is None
+            else self.config.override.functions(recipe_implementation)
+        )
+
+        api_implementation = APIImplementation()
+        self.api_implementation = (
+            api_implementation
+            if self.config.override.apis is None
+            else self.config.override.apis(api_implementation)
+        )
+
+    def is_error_from_this_recipe_based_on_instance(self, err: Exception) -> bool:
+        return isinstance(err, SuperTokensError) and (
+            isinstance(err, SuperTokensDashboardError)
+        )
+
+    def get_apis_handled(self) -> List[APIHandled]:
+        # Normally this array is used by the SDK to decide whether the recipe
+        # handles a specific API path and method and then returns the ID.
+
+        # However, for the dashboard recipe this logic is fully custom and handled inside the
+        # `return_api_id_if_can_handle_request` method of this class. Since this array is never
+        # used for this recipe, we simply return an empty array.
+
+        return []
+
+    async def handle_api_request(
+        self,
+        request_id: str,
+        request: BaseRequest,
+        path: NormalisedURLPath,
+        method: str,
+        response: BaseResponse,
+    ) -> Optional[BaseResponse]:
+        api_options = APIOptions(
+            request,
+            response,
+            self.recipe_id,
+            self.config,
+            self.recipe_implementation,
+            self.get_app_info(),
+        )
+        # For these APIs we dont need API key validation
+        if request_id == DASHBOARD_API:
+            return await handle_dashboard_api(self.api_implementation, api_options)
+        if request_id == VALIDATE_KEY_API:
+            return await handle_validate_key_api(self.api_implementation, api_options)
+
+        # Do API key validation for the remaining APIs
+        api_function: Optional[
+            Callable[[APIInterface, APIOptions], Awaitable[Optional[BaseResponse]]]
+        ] = None
+        if request_id == USERS_LIST_GET_API:
+            api_function = handle_users_get_api  # type: ignore
+        if request_id == USERS_COUNT_API:
+            api_function = handle_users_count_get_api  # type: ignore
```

Comment:
why need type ignore? You should be able to get the types working here?

Reply:
Fixed. Thanks!